src/osp/unpack.h

```diff
@@ -22,26 +22,29 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-#include "scenarios.h"
+#pragma once
 
-#include "flight.h"
+#include <cassert>
+#include <iterator>
+#include <type_traits>
 
-namespace testapp::flight
+namespace osp
 {
 
-entt::any setup_scene()
+/**
+ * @brief Create a structured binding-compatible type from a contiguous
+ *        container. A c-array is used.
+ */
+template<std::size_t N, typename RANGE_T>
+constexpr auto& unpack(RANGE_T &rIn)
 {
-    entt::any sceneAny = entt::make_any<FlightScene>();
-    FlightScene &rScene = entt::any_cast<FlightScene&>(sceneAny);
-
-    rScene.m_nwtWorld = std::make_unique<ospnewton::ACtxNwtWorld>(1);
+    using ptr_t = decltype(rIn.data());
+    using type_t = std::remove_pointer_t<ptr_t>;
 
-    return sceneAny;
+    assert(N <= rIn.size());
+    return *reinterpret_cast<type_t(*)[N]>(std::data(rIn));
```

Comment:
i don't think you should be trying to cast the return value of std::data() to pointer-to-array.

What are you actually trying to accomplish?

Reply:
I'd agree this may be sketchy, but so far it works. This function should return a type that works with structured bindings (a C-array) to assign variable names to elements in a (contiguous) range, inspired by unpacking in Python.

i.e.
```cpp
std::vector<int> stuff = {24, 165, 5};
auto & [a, b, c] = osp::unpack<3>(stuff);
```